tests/e2e/specs/shopper/checkout-single-use-coupon.test.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Internal dependencies
+ */
+import { shopper } from '../../../utils';
+import { createCoupon } from '../../utils';
+import { SIMPLE_PRODUCT_NAME } from '../../../utils/constants';
+import { getOrderDetailsDiscountPathExpression } from '../../../utils/path-expressions';
+
+if ( process.env.WOOCOMMERCE_BLOCKS_PHASE < 2 )
+	// eslint-disable-next-line jest/no-focused-tests
+	test.only( `Skipping checkout tests`, () => {} );
+
+let couponCode;
+
+beforeAll( async () => {
+	couponCode = await createCoupon( { usageLimit: 1 } );
+	await shopper.block.emptyCart();
+} );
+
+afterAll( async () => {
+	await shopper.block.emptyCart();
+} );
+
+describe( 'Shopper → Checkout → Can apply single-use coupon once', () => {
+	it( 'allows checkout to apply single-use coupon once', async () => {
+		await shopper.goToShop();
+		await shopper.addToCartFromShopPage( SIMPLE_PRODUCT_NAME );
+		await shopper.block.goToCheckout();
+
+		// Apply the single-use coupon.
+		await shopper.block.applyCouponFromCheckout( couponCode );
+
+		// Verify that the discount had been applied correctly on the checkout page.
+		// Selectors
```

Comment:
I'd probably remove a few of these inline comments, as I don't see an added value by having them.

Reply:
Thank you! I've cleaned up the comments! 👍